src/main/java/ch/njol/skript/expressions/ExprEntities.java

```diff
@@ -180,35 +181,14 @@ public boolean isLoopOf(String s) {
 					return false;
 				});
 		} else {
-			if (worlds == null && returnType == Player.class)
+			if (worlds == null && chunks == null && returnType == Player.class)
 				return super.iterator(e);
 
-			return new NonNullIterator<Entity>() {
-					private World[] ws = worlds == null ? Bukkit.getWorlds().toArray(new World[0]) : worlds.getArray(e);
-					private EntityData<?>[] ts = types.getAll(e);
-					private int w = -1;
-					@Nullable
-					private Iterator<? extends Entity> curIter = null;
-
-					@Override
-					@Nullable
-					protected Entity getNext() {
-						while (true) {
-							while (curIter == null || !curIter.hasNext()) {
-								w++;
-								if (w == ws.length)
-									return null;
-								curIter = ws[w].getEntitiesByClass(returnType).iterator();
-							}
-							while (curIter.hasNext()) {
-								Entity current = curIter.next();
-								for (EntityData<?> t : ts) {
-									if (t.isInstance(current))
-										return current;
-								}
-							}
-						}
-					}};
+			if (chunks != null) {
+				return Arrays.stream(EntityData.getAll(types.getArray(e), returnType, chunks.getArray(e))).iterator();
+			} else {
+				return Arrays.stream(EntityData.getAll(types.getAll(e), returnType, worlds != null ? worlds.getArray(e) : null)).iterator();
```

Comment:
Just a style nitpick — these lines are really long, perhaps it'd be a good idea to split them?

Reply:
If you take the middle bit out first and store it then you can do a single return statement with that value, which will make your line shorter to make blue happy. :)